### PR TITLE
Aggregator test uses framework namespace

### LINKED ISSUE #53478: Aggregator test namespace duplication
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
Aggregator test creates a new namespace 
https://github.com/kubernetes/kubernetes/blob/a5ad6b5ab6707ea411c9779beb789cdc8b4fee13/test/e2e/apimachinery/aggregator.go#L178-L191
instead of using existing test framework namespace
https://github.com/kubernetes/kubernetes/blob/a5ad6b5ab6707ea411c9779beb789cdc8b4fee13/test/e2e/apimachinery/aggregator.go#L171

The test should use framework namespace for better clean up and further development.

/sig api-machinery

### PR DESCRIPTION
**Pull request by roycaihw**
**What this PR does / why we need it**:
Remove namespace duplicate in aggregator test, using test framework namespace instead of creating a new one.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #53478 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

/sig api-machinery

### CODE REVIEW

File: test/e2e/apimachinery/aggregator.go [hunk 1]
```diff
@@ -159,36 +159,24 @@ func setupSampleAPIServerCert(namespaceName, serviceName string) *aggregatorCont
...
-			if strings.HasPrefix(err.Error(), "object is being deleted:") {
-				return false, nil
-			}
-			return false, err
-		}
-		namespace = got.Name
-		return true, nil
-	})
-	framework.ExpectNoError(err, "creating namespace %q", namespaceName)
+	// NOTE: use the namespace in test framework (normally user will create a namespace)
```
Comment (cheftako) [inline test/e2e/apimachinery/aggregator.go]: Please add comment to the effect of aggregated apis should generally be set up in there own namespace. As the test framework is setting up a new namespace, we are just using that.

**Comment by roycaihw** [comment]
updated comment and squashed @cheftako

**Review by cheftako — COMMENTED** [review]
Watch out for the namespace usage in validateErrorWithDebugInfo.

**Review by cheftako — COMMENTED** [review]
Also be careful of the namespace usage in cleanTest.

**Comment by roycaihw** [comment]
Thanks for catching those! Updated. @cheftako

**Review by cheftako — APPROVED** [review]
/lgtm

File: test/e2e/apimachinery/aggregator.go [hunk 2]
```diff
@@ -202,7 +190,7 @@ func TestSampleAPIServer(f *framework.Framework, image, namespaceName string) {
 			"tls.key": context.apiserverKey,
 		},
 	}
-	_, err = client.CoreV1().Secrets(namespace).Create(secret)
+	_, err := client.CoreV1().Secrets(namespace).Create(secret)
```
Comment (roycaihw) [inline test/e2e/apimachinery/aggregator.go]: Fix for compile failure: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/53479/pull-kubernetes-e2e-gce-bazel/33539/

**Review by cheftako — APPROVED** [review]
/lgtm

**Comment by tallclair** [comment]
Any updates on this?

File: test/e2e/apimachinery/aggregator.go [hunk 3]
```diff
@@ -159,36 +159,23 @@ func setupSampleAPIServerCert(namespaceName, serviceName string) *aggregatorCont
 
 // A basic test if the sample-apiserver code from 1.7 and compiled against 1.7
 // will work on the current Aggregator/API-Server.
-func TestSampleAPIServer(f *framework.Framework, image, namespaceName string) {
+func TestSampleAPIServer(f *framework.Framework, image string) {
 	By("Registering the sample API server.")
-	cleanTest(f, true)
```
Comment (roycaihw) [inline test/e2e/apimachinery/aggregator.go]: @cheftako Update to keep framework namespace alive.

**Review by cheftako — APPROVED** [review]
/lgtm

File: test/e2e/apimachinery/aggregator.go [hunk 4]
```diff
@@ -68,37 +68,24 @@ var _ = SIGDescribe("Aggregator", func() {
...
 
-	namespace := "sample-system"
+	namespace := f.Namespace.Name
 	client := f.ClientSet
 	_ = client.ExtensionsV1beta1().Deployments(namespace).Delete("sample-apiserver", nil)
 	_ = client.CoreV1().Secrets(namespace).Delete("sample-apiserver-secret", nil)
 	_ = client.CoreV1().Services(namespace).Delete("sample-api", nil)
 	_ = client.CoreV1().ServiceAccounts(namespace).Delete("sample-apiserver", nil)
 	_ = client.RbacV1beta1().RoleBindings("kube-system").Delete("wardler-auth-reader", nil)
-	_ = client.CoreV1().Namespaces().Delete(namespace, nil)
```
Comment (roycaihw) [inline test/e2e/apimachinery/aggregator.go]: @cheftako Update to avoid potential race condition.

**Comment by roycaihw** [comment]
Looks like the tests are all green now. Ping @cheftako @janetkuo for approval.

**Review by cheftako — APPROVED** [review]
/lgtm

**Review by janetkuo — APPROVED** [review]

**Pull request merged**